compaction/compaction.cc

```diff
@@ -481,6 +481,7 @@ class compaction {
     utils::observable<> _stop_request_observable;
 private:
     compaction_data& init_compaction_data(compaction_data& cdata, const compaction_descriptor& descriptor) const {
+        cdata.the_compaction = static_cast<const void *>(this);
```

Comment:
I don't think this is needed. `_cdata` is a member of `class compaction`, which is a virtual class. It is trivial to iterate over all existing `compaction` objects in memory and find the one which has the known value for `_cdata`. This can be also done with finding references to a given data instance with `--resolve`.